crates/iceberg/src/spec/manifest.rs

```diff
@@ -1311,511 +1311,452 @@ mod tests {
+                            2,
+                            "v_int",
+                            Type::Primitive(PrimitiveType::Int),
+                        )),
+                        Arc::new(NestedField::optional(
+                            3,
+                            "v_long",
+                            Type::Primitive(PrimitiveType::Long),
+                        )),
+                        Arc::new(NestedField::optional(
+                            4,
+                            "v_float",
+                            Type::Primitive(PrimitiveType::Float),
+                        )),
+                        Arc::new(NestedField::optional(
+                            5,
+                            "v_double",
+                            Type::Primitive(PrimitiveType::Double),
+                        )),
+                        Arc::new(NestedField::optional(
+                            6,
+                            "v_varchar",
+                            Type::Primitive(PrimitiveType::String),
+                        )),
+                        Arc::new(NestedField::optional(
+                            7,
+                            "v_bool",
+                            Type::Primitive(PrimitiveType::Boolean),
+                        )),
+                        Arc::new(NestedField::optional(
+                            8,
+                            "v_date",
+                            Type::Primitive(PrimitiveType::Date),
+                        )),
+                        Arc::new(NestedField::optional(
+                            9,
+                            "v_timestamp",
+                            Type::Primitive(PrimitiveType::Timestamptz),
+                        )),
+                        Arc::new(NestedField::optional(
+                            10,
+                            "v_decimal",
+                            Type::Primitive(PrimitiveType::Decimal {
+                                precision: 36,
+                                scale: 10,
+                            }),
+                        )),
+                        Arc::new(NestedField::optional(
+                            11,
+                            "v_ts_ntz",
+                            Type::Primitive(PrimitiveType::Timestamp),
+                        )),
+                    ])
+                    .build()
+                    .unwrap(),
+                partition_spec: PartitionSpec {
+                    spec_id: 0,
+                    fields: vec![],
+                },
+                content: ManifestContentType::Data,
+                format_version: FormatVersion::V2,
+            },
+            entries: vec![
+                ManifestEntry {
+                    status: ManifestStatus::Added,
+                    snapshot_id: Some(0),
+                    sequence_number: Some(1),
+                    file_sequence_number: Some(1),
```

Comment:
Typically when a new manifest-entry is being added, the snapshot-id and sequence-numbers are not set and inherited from the snapshot itself.

Reply:
Fixed. In v2 by default it's `None`, and in v1 it should be `0`.